068-access-policies-in-jwt.md

```diff
@@ -0,0 +1,153 @@
+# Pass Access Policies (ACLs) in JWT
+
+This proposal addresses the possibility of passing access policies in the JWT tokens.
+
+## Current situation
+
+Currently, authorization is supported via [Keycloak](https://www.keycloak.org/).
+Keycloak must be deployed, and the identities must be managed there.
+This makes using Strimzi in larger systems more complicated.
+Most users use their own identity management systems.
+In order to integrate it with Strimzi, they would have to duplicate the identities in Keycloak.
+This solution doesn't follow the single source of truth architecture.
+It also requires to run a redundant service - Keycloak.
+
+## Motivation
+
+Having a generic solution is always preferable as it allows for less effort to be maintained.
+While we cannot achieve a perfect generic solution due to the nature of OAuth2, and how different
+identity providers are implemented, we can strive to create a solution as generic as possible.
+
+Passing additional data in the JWT seems to be supported by the majority of identity providers.
+Therefore, we can use this to pass the access policies in the JWT.
+This solution will work in most cases and with most identity providers, however it will not
+scale to infinity.
+It will, however, be decoupled from Keycloak and therefore be more generic.
+
+## Proposal
+
+Identity providers support passing custom fields in the JWT token.
+Some providers allow the data type to be string, while others allow a list of strings.
+There are not many (if any) identity providers that allow a list of objects.
+
+Therefore, in this proposal we allow the ACLs to be passed as either.
+e.g. for a list of strings:
+```json
+{
+  "<acls>": [
```

Comment:
Wouldn't that make more sense as `"acls"` rather than `"<acls>"`?

Reply:
Yes, I originally had it `<acls>` to indicate it's a configurable parameter, but I mention this in the text, so I can change this to `acls`.